src/Raven.Server/Documents/Queries/Results/MapReduceQueryResultRetriever.cs

```diff
@@ -89,7 +91,7 @@ public override unsafe Document DirectGet(ref RetrieverInput retrieverInput,stri
             BlittableJsonReaderObject result;
             if (retrieverInput.IsLuceneDocument() == false)
             {
-                retrieverInput.CoraxEntry.GetFieldReaderFor(FieldsToFetch.IndexFields.Count + 1).Read(out var binaryValue);
+                retrieverInput.CoraxEntry.GetFieldReaderFor(retrieverInput.KnownFields.StoredJsonPropertyOffset).Read(out var binaryValue);
```

Comment:
You use `StoredJsonPropertyOffset` which does `.Count - 1` under the covers. Although here you had `.Count + 1`. Please double check that.

Reply:
Okay, I think it's good to describe it well here because it's tricky.

At the beginning of the Corax integration, we created known fields on `IndexFields` (the numeration was from 1 up N, with 0 reserved for `id()`). Since we didn't have something like stored() in the mapping, it was implied that the fields were just after all the known fields (which is why I used `+1`). After that, we introduced `IndexFieldsMapping` (since we added analyzers, changed modes during querying/writing, etc.), and we added a `stored()` field to the mapping. This is why it's correct. Furthermore, all MapReduce tests where we do projection rely on it, so it has been well tested.